### PR TITLE
ENH: Add handling for DatetimeTZ to discover.

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

**Comment by ssanderson** [comment]
ping @llllllllll

File: odo/backends/tests/test_pandas.py [hunk 1]
```diff
@@ -81,3 +85,35 @@ def test_categorical_pandas():
...
+            ('naive', pd.date_range('2014', periods=5)),
+            ('Europe/Moscow', pd.date_range('2014', periods=5, tz='Europe/Moscow')),
+            ('UTC', pd.date_range('2014', periods=5, tz='UTC')),
+            ('US/Eastern', pd.date_range('2014', periods=5, tz='US/Eastern')),
+        ])
+    )
+
+    assert_dshape_equal(
+        discover(df),
+        5 * Record([
```
Comment (llllllllll) [inline odo/backends/tests/test_pandas.py]: You can just use a record 'literal' here, `R['key': type, ...]`

File: odo/backends/tests/test_pandas.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
 from __future__ import absolute_import, division, print_function
 
+from collections import OrderedDict
 from datetime import datetime, timedelta
+from distutils.version import StrictVersion
 
-from datashape import discover, float64, dshape, Record, Categorical, int64
+from datashape import (
+    discover, float64, dshape, Record, Categorical, DateTime, Option, int64
```
Comment (llllllllll) [inline odo/backends/tests/test_pandas.py]: could you lay this out vertically?

**Comment by llllllllll** [comment]
2 style comments but lgtm

**Comment by ssanderson** [comment]
@llllllllll style nits addressed.

**Pull request merged**